docs/2-deploying/1-automated-previews.md

```diff
@@ -41,42 +41,49 @@ jobs:
+
+This configuration takes advantage of GitLab environments in order to give you better control and visibility into what environments exist and what's deployed to them. On PR creation, both a GitLab and Architect environment will be created. The component specified in the repository will be registered with the Architect Cloud and deployed to the environment. When the PR is either merged or closed, the GitLab environment will be automatically deleted and the component deployed to the environment in the Architect Cloud will be destroyed.
 
 ```yaml
-preview:
-  stage: deploy
+stages:
+  - preview
+
+default:
   image: docker:latest
   services:
     - docker:dind
-  only:
-    - merge_requests
-  variables:
-    ARCHITECT_ACCOUNT: my-account
-    PREVIEW_PLATFORM: architect
-    PREVIEW_TAG: $CI_MERGE_REQUEST_SOURCE_BRANCH_NAME
   before_script:
     - apk add --update npm git
     - apk add yq --repository=http://dl-cdn.alpinelinux.org/alpine/edge/community
-    - npm i -g @architect-io/cli --unsafe-perm
+    - npm install -g @architect-io/cli@0.8 --unsafe-perm
     - architect login -e $ARCHITECT_EMAIL -p $ARCHITECT_PASSWORD
-  script:
-    # Use the yq library to extract the name of the component
-    - export FULL_COMPONENT_NAME=$(yq r ./architect.yml name)
-    # Remove the account name from the component name lifted from the manifest
-    - export COMPONENT_NAME=${FULL_COMPONENT_NAME##*/}
-    # Register the component
-    - architect register . -t $PREVIEW_TAG
-    # Create a preview environment if one doesn't already exist
-    - architect env:create $COMPONENT_NAME-$CI_MERGE_REQUEST_ID -a $ARCHITECT_ACCOUNT --platform $PREVIEW_PLATFORM || echo "Preview environment already exists. Deploying to it now."
-    # (optional) extract the interfaces from the manifest so we can automatically expose them
-    - >
-      export INTERFACES=""
-      for i in $(yq r --printMode p ./src/product-catalog/architect.yml "interfaces.*"); do
-        export INTERFACES="$INTERFACES-i ${i#"interfaces."}:${i#"interfaces."} "
-      done
-    # Deploy the component to the preview environment
-    - architect deploy $FULL_COMPONENT_NAME:$PREVIEW_TAG -e $COMPONENT_NAME-$CI_MERGE_REQUEST_ID -a $ARCHITECT_ACCOUNT --auto_approve $INTERFACES
+
+deploy_preview:
+  stage: preview
+  script: |
+    architect register architect.yml -t $CI_MERGE_REQUEST_ID
+    architect environment:create $CI_MERGE_REQUEST_ID -a $ARCHITECT_ACCOUNT --platform $ARCHITECT_PLATFORM || true
+    architect deploy --auto_approve -a $ARCHITECT_ACCOUNT -e $CI_MERGE_REQUEST_ID $ARCHITECT_COMPONENT_NAME:$CI_MERGE_REQUEST_ID $ARCHITECT_DEPLOY_FLAGS
+  environment:
+    name: architect/preview/$CI_MERGE_REQUEST_ID
+    url: https://app.architect.io/ryan-cahill-a0/environments/$CI_MERGE_REQUEST_ID/
+    on_stop: destroy_preview
+  rules:
+    - if: $CI_MERGE_REQUEST_ID
```

Comment:
A little torn, but the point of the ARCHITECT_ACCOUNT env is that we don't have to specify the -a flag.
I think I'd prefer the following:

```yml
variables:
  ARCHITECT_ACCOUNT: my-account
  ARCHITECT_PLATFORM: my-platform
  
deploy_preview:
  stage: preview
  script: |
    architect register architect.yml -t $ARCHITECT_ENVIRONMENT
    architect environment:create || echo 'Environment already exists'
    architect deploy $ARCHITECT_COMPONENT_NAME:$ARCHITECT_ENVIRONMENT --auto_approve
  environment:
    name: architect/$ARCHITECT_ENVIRONMENT
    on_stop: destroy_preview
  variables:
    ARCHITECT_ENVIRONMENT: preview-$CI_MERGE_REQUEST_ID
  rules:
    - if: $CI_MERGE_REQUEST_ID
    
destroy_preview:
  stage: preview
  script: |
    architect destroy -c $ARCHITECT_COMPONENT_NAME:$ARCHITECT_ENVIRONMENT --auto_approve
    architect environment:destroy --auto_approve
  environment:
    name: architect/$ARCHITECT_ENVIRONMENT
    action: stop
  variables:
    ARCHITECT_ENVIRONMENT: preview-$CI_MERGE_REQUEST_ID
  rules:
    - if: $CI_MERGE_REQUEST_ID
      when: manual
```

Reply:
I agree that your way is better, but I also think that when people are referring to the docs, they might want more verbose examples. How in this case is someone supposed to know which command(s) `ARCHITECT_ACCOUNT` is used for? I'd rather make things more clear

Reply:
We could add yaml comments at the top that dictate what each variables does. 

It also might make sense to add more logging in the cli cmds when we implicitly pickup values.
ex:
```bash
$ architect environment:create
> Using account context: tjhiggins 
> Using platform context: my-platform
```

Reply:
Sure, I like that. I'll do the same for the account, platform, and environment flags

Reply:
@tjhiggins I like that idea too. Let's just be sure to be a tad more verbose like:

```sh
$ architect environment:create
> Using account from environment variables: tjhiggins
> Using platform from environment variables: my-platform
```